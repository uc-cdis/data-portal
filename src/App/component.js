import React from 'react';
import { withBoxAndNav, withAuthTimeout } from '../utils';
import Relay from 'react-relay'
import styled from 'styled-components';
import { Link } from 'react-router';

const ProjectLink = styled(Link)`
  cursor: pointer;
  li;
  background: ghostwhite;
  padding: 5px;
  border-radius: 5px;
  display: inline-block;
  margin-bottom: 1em;
  margin-right: 1em;
  color: #8a6d3b;
  &:hover,
  &:focus,
  &:active {
    border: 1px solid #8a6d3b;
    color: #8a6d3b;
  }
`;

class SubmissionComponent extends React.Component {
  static propTypes = {
    viewer: React.PropTypes.object,
  };
  render () {
    return (
      <div>
        <h3>Submission projects</h3>
        <ul>
          {this.props.viewer.project.map((p)=> <ProjectLink to={'/'+p.project_id} key={p.project_id}>{p.project_id}</ProjectLink>)}
        </ul>
      </div>
    )
  }
}

let Submission = Relay.createContainer(
  withBoxAndNav(withAuthTimeout(SubmissionComponent)),
  {
    fragments: {
      viewer: () => Relay.QL`
          fragment on viewer {
              project (first: 10000) {
                  project_id
                  code
              }
          }
      `,
    },
  },
);
export default Submission;
