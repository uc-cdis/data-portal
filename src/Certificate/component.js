import React, { Component } from 'react';
import { Form, FormError, RadioGroup, Radio } from 'react-form';
import Nav from '../Nav/component.js';
import { submitForm, updateForm } from './actions';
import { connect } from 'react-redux';
import { QuestionItem, OptionBullet, SubmitButton, Tooltip } from './style';
import * as constants from './constants';

class Question extends Component {
  render() {
    return (
      <QuestionItem>
        <div>
          <span>{this.props.content.question}</span>
          <Tooltip className="fui-question-circle">
            <p>{this.props.content.hint}</p>
          </Tooltip>
          { this.props.showErrors && <FormError field={this.props.content.id} /> }
        </div>
        <RadioGroup showErrors={false} field={this.props.content.id}>
          {this.props.content.options.map((option, i) => <OptionBullet key={i}><Radio onChange={this.props.onChange} name={this.props.content.id} value={option} />{option}</OptionBullet>,
          )}
        </RadioGroup>
      </QuestionItem>
    );
  }
}

class CertificateComponent extends Component {
  constructor() {
    super();
    this.state = { display_error: false };
    // required to be able to pass to child
    this.hideError = this.hideError.bind(this);
  }
  validateForm(values) {
    // update redux store
    if (Object.keys(values).length > 0) {
      this.props.onUpdateForm(values);
    }
    const result = {};
    for (const item of constants.certificate_form) {
      result[[item.id]] = values[item.id] === item.options[item.answer] ? undefined : 'wrong answer';
    }
    return result;
  }
  // hide errors when user is updating the form
  hideError() {
    this.setState({ display_error: false });
  }

  // show errors when user hits submit button
  showError() {
    this.setState({ display_error: true });
  }
  render() {
    return (
      <div>
        <h4>{constants.title}</h4>
        <Form
          onSubmit={(values) => { this.props.onSubmitForm(values); }}
          validate={values => this.validateForm(values)}
          defaultValues={this.props.certificate.certificate_result}
        >

          {({ errors, submitForm }) => (
            <form>
              {
                constants.certificate_form.map((item, i) => <Question content={item} onChange={this.hideError} index={i} key={i} showErrors={this.state.display_error} />)
              }
              <SubmitButton type="submit" onClick={() => { submitForm(); this.showError(); }}>Submit</SubmitButton>
            </form>
          )}
        </Form>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user,
  certificate: state.certificate,
});

const mapDispatchToProps = dispatch => ({
  onUpdateForm: data => dispatch(updateForm(data)),
  onSubmitForm: data => dispatch(submitForm(data)),
});

const Certificate = connect(mapStateToProps, mapDispatchToProps)(CertificateComponent);
export default Certificate;
