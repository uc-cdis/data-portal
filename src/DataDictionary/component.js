import React from 'react';
import Nav from '../Nav/component.js';
import { Box, cube, Table, TableData, TableRow, TableHead } from '../theme.js';
import { connect } from 'react-redux';
import styled from 'styled-components';
import * as constants from './constants';
import { Link } from 'react-router';

const TableBullet = ({ node, description }) => (
  <TableRow>
    <TableData>
      <Link to={`/dd/${node}`}> {node} </Link>
    </TableData>
    <TableData right>
      {description}
    </TableData>
  </TableRow>
);


const CategoryTable = ({ dictionary, nodes, category }) => (
  <Table>
    <TableHead>
      <TableRow>
        <TableData first_cr>
          {category}
        </TableData>
      </TableRow>
    </TableHead>

    <tbody>
      {nodes.map(node => <TableBullet node={node} key={node} description={dictionary[node].description} />)}
    </tbody>
  </Table>
);

const DataDictionaryViewer = ({ submission }) => {
  const filterCategories = (dictionary, node_types) => {
    const categories = {};
    for (const node in node_types) {
      const category = dictionary[node_types[node]].category;
      if (category == undefined) {
        continue;
      }
      if (categories.hasOwnProperty(category)) {
        categories[category].push(node_types[node]);
      } else {
        categories[category] = [node_types[node]];
      }
    }
    return categories;
  };
  const categories = filterCategories(submission.dictionary, submission.node_types);
  return (
    <div>
      <h3> Data Dictionary Viewer </h3>
      <p>{constants.subHeader}</p>
      <Link to={'/dd/graph'}> Explore dictionary as a graph </Link>
      {Object.keys(categories).map(category =>
        <CategoryTable dictionary={submission.dictionary} key={category} nodes={categories[category]} category={category} />)}
    </div>
  );
};

const mapStateToProps = state => ({
  submission: state.submission,
});

const mapDispatchToProps = dispatch => ({
});

const DataDictionary = connect(mapStateToProps, mapDispatchToProps)(DataDictionaryViewer);
export default DataDictionary;
