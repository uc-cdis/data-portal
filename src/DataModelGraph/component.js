import React from 'react';
import { connect } from 'react-redux';
import { getCounts } from './actions';
import { submissionapi_path } from '../localconf';
import { button } from '../theme';
import styled from 'styled-components';
import { createNodesAndEdges } from '../utils'
import { create_dm_graph } from './GraphCreator'
import * as d3 from "d3";

export const ToggleButton = styled.a`
  border: 1px solid darkslategray;
  color: darkslategray;
  ${button};
  position:absolute;
  top:15px;
  left:20px;
  &:hover,
  &:active,
  &:focus {
    color: black;
    border-color: black;
  }
`;


class DataModelGraphComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      compact: undefined,
      full: undefined, 
      full_toggle: false
    }
    this.state.compact = createNodesAndEdges(props, this.state.full_toggle);
    this.handleClick = this.handleClick.bind(this);
  }
  componentWillReceiveProps(nextProps) {
    if (this.state.full_toggle) {
      this.state.full = createNodesAndEdges(nextProps, this.state.full_toggle);
    } else {
      this.state.compact = createNodesAndEdges(nextProps, this.state.full_toggle);
    }
  } 
  handleClick() {
    if (this.state.full_toggle) {
      this.setState(prevState => ({
        full_toggle: !prevState.full_toggle,
        compact: createNodesAndEdges(this.props,!this.state.full_toggle)
      }));
    } else {
      this.setState(prevState => ({
        full_toggle: !prevState.full_toggle,
        full: createNodesAndEdges(this.props,!this.state.full_toggle)
      }));
    }
  }
  render() {
    if (this.state.full_toggle) {
      if (this.state.full.nodes.length != 0 && 'count' in this.state.full.nodes[this.state.full.nodes.length-1]) {
        return (
          <div style={{position: "relative"}}>
            <ToggleButton onClick={this.handleClick}>Toggle view</ToggleButton>
            <CreateGraph nodes={this.state.full.nodes} edges={this.state.full.edges}/>
          </div>
        );
      } 
    } else {
      if (this.state.compact.nodes.length != 0 && 'count' in this.state.compact.nodes[this.state.compact.nodes.length-1]) {
        return (
          <div style={{position: "relative"}}>
            <ToggleButton onClick={this.handleClick}>Toggle view</ToggleButton>
            <CreateGraph nodes={this.state.compact.nodes} edges={this.state.compact.edges}/>
          </div>
        );
      }
    } 
    return null;
  }
}

class CreateGraph extends React.Component {
  componentDidMount() {
    create_dm_graph(this.props.nodes, this.props.edges)
  }
  componentDidUpdate() {
    create_dm_graph(this.props.nodes, this.props.edges)
  }
  render() {
    let nodes = this.props.nodes
    let edges = this.props.edges

    let root = "project"
    let queue = [];
    let layout = [];
    let placed = [];
    let layout_level = 0;

    queue.push(root);
    layout.push([root]);
    while(queue.length != 0) {
      let query = queue.shift(); //breadth first
      for (let i = 0; i < edges.length; i++) {
        if (edges[i].target == query || edges[i].target.name == query) {
          if ((layout[layout_level].indexOf(query)) != -1) {
            if (!layout[layout_level+1]) {
              layout[layout_level+1] = [];
            } 
          } else {
            layout_level += 1;
            if (!layout[layout_level+1]) {
              layout[layout_level+1] = [];
            } 
          }
          queue.push(edges[i].source);
          if ((layout[layout_level+1].indexOf(edges[i].source) == -1) && (placed.indexOf(edges[i].source) == -1)) {
            layout[layout_level+1].push(edges[i].source);
            placed.push(edges[i].source);
          }
        }
      }
      placed.push(query);
    }

    for (let i = 0; i < layout.length; i++) {
      for (let j = 0; j < layout[i].length; j++) {
        for (let k = 0; k < nodes.length; k++) {
          if (nodes[k].name == layout[i][j]) {
            nodes[k].position = [(j+1)/(layout[i].length+1),(i+1)/(layout.length+1)];
            break;
          }
        }
      }
    }

    let min_x_pos = Math.round(1/d3.extent(nodes.map((node) => node.position[0]))[0])
    let min_y_pos = Math.round(1/d3.extent(nodes.map((node) => node.position[1]))[0])

    let padding = 25, 
      radius = 60,
      legend_width=125,
      width = min_x_pos * radius * 2 + legend_width, 
      height = min_y_pos * radius * 2 + padding;

    const divStyle = {
      height: height,
      backgroundColor: "#f4f4f4",
      marginLeft: "auto",
      marginRight: "auto",
    }
    return (
      <div style={divStyle}>
        <svg id="data_model_graph" height={height} width={width}>
        </svg>
      </div>
      );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    'dictionary': state.submission.dictionary,
    'counts_search': state.submission.counts_search,
    'links_search': state.submission.links_search,
  };
}

const mapDispatchToProps = (dispatch) => {
  return {
    onGetCounts: (type, project) => dispatch(getCounts(type, project)),
  };
}
const DataModelGraph = connect(mapStateToProps, mapDispatchToProps)(DataModelGraphComponent);
export default DataModelGraph;
