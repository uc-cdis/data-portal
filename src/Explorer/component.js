import React, {Component, PropTypes} from 'react';
import {connect} from 'react-redux';
import {CheckBoxGroup} from "../components/CheckBox";
import styled from 'styled-components';

const TableExplorer = ({})=>{
  return(
    <div>
    TableExplorer
    </div>
  )
};

class Sidebar extends Component {
  static propTypes = {
    projects: PropTypes.object,
    file_formats: PropTypes.array,
  };

  state = {
    form: {},
  };

  render(){
    return(
      <div>
      <CheckBoxGroup listItems={Object.values(this.props.projects)} title="Projects"
                group_name="Project" onChange={() => alert("checked")}/>
      <CheckBoxGroup listItems={this.props.file_formats} title="File Formats"
                group_name="file_formats" onChange={() => alert("checked")} />
      </div>
    );
  };
}

class ExplorerComponent extends Component {
  static propTypes = {
    submission: PropTypes.object,
  };

  aggregateFileFormats = (dictionary) =>{
    let file_formats = new Set();
    console.log(file_formats)
    if(dictionary == 'undefined'){
      return(file_formats);
    }
    for(let node in dictionary){
      if(dictionary[node].hasOwnProperty('category') && dictionary[node]['category'] == 'data_file'){
        if(dictionary[node]['properties']['data_type'].hasOwnProperty('enum')){
          for(let file_format of dictionary[node]['properties']['data_type']['enum']){
            if(!file_formats.has(file_format)){
              file_formats.add(file_format);
            }
          }
        }
      }
    }
    return(file_formats);
  };

  state = {
    projects: this.props.submission.projects,
<<<<<<< HEAD
  };
=======
    selected_projects: null,
    file_formats: this.aggregateFileFormats(this.props.submission.dictionary),
    dictionary:this.props.submission.dictionary,

  }
>>>>>>> de94a34... feat(explorer): generate list of file_formats


  render(){
    return(
      <div>
      <Sidebar projects={this.state.projects} file_formats={Array.from(this.state.file_formats.values())} />
      <TableExplorer />
      </div>
    )
  };

}

const mapStateToProps = (state) => {
    return{
      'submission': state.submission,
    }
};


const mapDispatchToProps = (dispatch) =>{
    return{
    }
};

let Explorer = connect(mapStateToProps, mapDispatchToProps)(ExplorerComponent);

export default Explorer;
