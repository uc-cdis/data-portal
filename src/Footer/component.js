import styled, { css } from 'styled-components';
import { Link } from 'react-router';
import React from 'react';

const FooterSection = styled.footer`
  position: absolute;
  text-align: center;
  bottom: 0px;
  left: 0px;
  right: 0px;
  width: 100%;
  background-color: rgba(0, 0, 0, 0.87);
  z-index: 1000;
`;

const Dictionary = styled(Link)`
  padding: 5px;
  span {
    margin-right: 5px;
  }
`;
const NavRight = styled.nav`
  width: 100%;
  padding: 10px 100px;
  color: white;
`;

const Footer = () => (
  <FooterSection>
    <NavRight>
      <ul>
          <Dictionary to='/dd'><span className='fui-bookmark'></span>View dictionary</Dictionary>
      </ul>
    </NavRight>
  </FooterSection>

)

export default Footer;
