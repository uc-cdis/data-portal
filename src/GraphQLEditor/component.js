import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { fetchGraphQL } from '../actions';
import GraphiQL from 'graphiql';
import { buildClientSchema } from 'graphql/utilities';

export const Title = styled.h2`
  display: inline-block;
  vertical-align: middle;
  margin: 15px 0px;
  margin-right: 0.5em;
`;

export const GraphBox = styled.div`
  height: 100vh;
`;


let parameters = {};

const ProjectSubmissionComponent = ({schema}) => {
  let graphqlSchema = buildClientSchema(schema.data);
  let editQuery = (newQuery) => {
    parameters.query = newQuery;
  };
  let editVariables = (newVariables) => {
    parameters.variables = newVariables;
  };

  return (
    <div>
      <GraphBox id="graphiql">
        <Title>Query graph</Title>
        <GraphiQL
          fetcher={fetchGraphQL}
          schema={graphqlSchema}
          query={parameters.query}
          variables={parameters.variables}
          onEditQuery={editQuery}
          onEditVariables={editVariables} />
      </GraphBox>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    'schema': state.graphiql.schema
  }
};

const mapDispatchToProps = (dispatch) => ({
});

const GraphQLQuery = connect(mapStateToProps, mapDispatchToProps)(ProjectSubmissionComponent);
export default GraphQLQuery;
