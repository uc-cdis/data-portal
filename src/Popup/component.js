import React from 'react';
import styled, {css} from 'styled-components';
import Highlight from 'react-highlight';
import AceEditor from 'react-ace';
import { logoutAPI } from '../actions';
import { connect } from 'react-redux';
import { browserHistory } from 'react-router';

const button = css`
  display: inline-block;
  float:right;
  cursor: pointer;
  margin-left: 1em;
`;
export const Button = styled.a`
  ${button};
`;
const CancelButton = styled.a`
  color: ${props => props.theme.dark_gray};
  &:hover,
  &:active,
  &:focus {
    color: ${props => props.theme.mid_gray};
  }
  ${button};
`;

export const PopupMask = styled.section`
  z-index: 100;
  position: fixed;
  overflow-y: scroll;
  top: 0px;
  width: 100%;
  height: 100%;
  left: 0;
  background: rgba(223, 223, 223, 0.47);
  padding-top: 10em;
`;

const PopupBox = styled.section`
  width: 50%;
  overflow-y: scroll;
  background: white;
  margin: auto;
  padding: 2em;
  overflow: hidden;
`;

const Code = styled(Highlight)`
  font-size: 0.8em;
`;
const Message = styled.div`
  font-size: 1em;
`;

const SavePopupBox = styled.section`
  width: 80%;
  overflow-y: scroll;
  background: white;
  margin: auto;
  padding: 2em;
  overflow: hidden;
`;

export const Popup = ({message, code, error, closeText, cancelText, confirmText, onClose, onCancel, onConfirm}) => {
  return (
    <PopupMask>
      <PopupBox>
        <Message>
          <div>{message}</div>
          {code &&
          <Code className='json'> {code} </Code>
          }
          {error &&
          <Code className='json'> {error} </Code>
          }
        </Message>
        {onClose &&
          <CancelButton onClick={onClose}>{closeText ? closeText : "close"}</CancelButton>
        }
        {onConfirm &&
          <Button onClick={onConfirm}>{confirmText ? confirmText : "confirm"}</Button>
        }
        {onCancel &&
          <CancelButton onClick={onCancel}>{cancelText ? cancelText : "cancel"}</CancelButton>
        }
      </PopupBox>
    </PopupMask>
  )
};

export const saveToFile = (savingStr, filename) => {
  let FileSaver = require('file-saver');
  let blob = new Blob([savingStr], {type: "text/plain;charset=utf-8"});
  FileSaver.saveAs(blob, filename);
};

const timeoutPopupMapState = (state) => {
  return {
    'auth_popup': state.popups.auth_popup,
  }
};

const timeoutPopupMapDispatch = (dispatch) => {
  return {}
};

const goToLogin = () => {
    browserHistory.push('/login');
    // Refresh the page.
    window.location.reload(false);
};

export const AuthTimeoutPopup = connect(timeoutPopupMapState, timeoutPopupMapDispatch)(({auth_popup, onConfirmDoLogout}) => {
  if (auth_popup) {
    return <Popup message={'Your session has expired or you are logged out. Please log in to continue.'} confirmText='go to login' onConfirm={goToLogin} />
  }
  return (null)
});

let copy = require('clipboard-plus');
export const SavePopup = ({message, display, savingStr, error, onClose, filename}) => {
  return (
    <PopupMask>
      <SavePopupBox>
        <Message>
          {
            !error &&
            <div>{message}</div>
          }
          {
            display &&
            <div>
              <table width="100%">
                <tbody>
                  <tr><th>Access id</th><th>Secret key</th></tr>
                  <tr><td width="40%">{display['access_key']}</td><td width="60%">{display['secret_key']}</td></tr>
                </tbody>
              </table>
            </div>
          }
          {error &&
            <Code className='plaintext'> {error} </Code>
          }
        </Message>
        {onClose &&
          <CancelButton onClick={onClose}>close</CancelButton>
        }
        {
          savingStr &&
          <Button onClick={() => saveToFile(savingStr, filename)}>Save as</Button>
        }
        {
          savingStr &&
          <Button onClick={() => copy(savingStr)}>Copy</Button>
        }
      </SavePopupBox>
    </PopupMask>
  )
};
