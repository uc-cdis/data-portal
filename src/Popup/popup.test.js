import React from 'react';
import { Popup } from './component';
import renderer from 'react-test-renderer';


test('Test Popup', () => {
  const onClose = () => {
    console.log('closed');
  };
  const onCancel = () => {
    console.log('canceled');
  };
  const onConfirm = () => {
    console.log('confirmed');
  };
  // react-hightlight doesn't work with jest snapshot, so code and error is not passed
  const component = renderer.create(
    <Popup message="test" onConfirm={onConfirm} onCancel={onCancel} onClose={onClose} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
