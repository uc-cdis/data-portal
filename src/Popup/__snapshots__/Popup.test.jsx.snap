// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Popup 1`] = `
<section
  className="sc-htpNat dDTzPz"
>
  <section
    className="sc-bxivhb iZMisd"
  >
    <div
      className="sc-EHOje LUNcQ"
    >
      <div>
        test
      </div>
      
      
      
    </div>
    <a
      className="sc-bwzfXH eBwlJa"
      id="cd-popup__button_close"
      onClick={[Function]}
      role="button"
    >
      close
    </a>
    <a
      className="sc-bdVaJa hiblcO"
      id="cd-popup__button_confirm"
      onClick={[Function]}
      role="button"
    >
      confirm
    </a>
    <a
      className="sc-bwzfXH eBwlJa"
      id="cd-popup__button_cancel"
      onClick={[Function]}
      role="button"
    >
      cancel
    </a>
  </section>
</section>
`;
